internal/plugins/scorecard/init.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	"sigs.k8s.io/yaml"
+
+	"github.com/operator-framework/operator-sdk/internal/scaffold/kustomize"
+	"github.com/operator-framework/operator-sdk/internal/scorecard"
+	"github.com/operator-framework/operator-sdk/pkg/apis/scorecard/v1alpha3"
+)
+
+// kustomization.yaml file template for the scorecard componentconfig. This should always be written to
+// config/scorecard/kustomization.yaml since it only references files in config.
+const scorecardKustomizationTemplate = `resources:
+{{- range $i, $path := .ResourcePaths }}
+- {{ $path }}
+{{- end }}
+patchesJson6902:
+{{- range $i, $patch := .JSONPatches }}
+- path: {{ $patch.Path }}
+  target:
+    group: {{ $patch.Target.Group }}
+    version: {{ $patch.Target.Version }}
+    kind: {{ $patch.Target.Kind }}
+    name: {{ $patch.Target.Name }}
+{{- end }}
```

Comment:
Why we do not it as the watches and the other features in kb and helm as well? 

IMO we should use the default kubebuilder markers instead of that. E.g `# +kubebuilder:scaffold:scorecard`. I do not think that is following the current design. Note that kb has a lot of helpers for the markers which indeed let us mock the tests which will not be possible to be used here.

Reply:
Because we would be defining our own struct for a kustomize type. This type may exist in the kustomize codebase, but I'd rather not import kustomize libraries if we don't have to.